### DISCUSSION TITLE
How reliable is the AFK module?

### DISCUSSION THREAD

**Discussion by Culpeo** [ActivityWatch/activitywatch#545] [Q&A] [answered]
Hej,

I downloaded ActivityWatch (Windows Installer v0.10.0) yesterday and played a bit around with it - you really did a great job so far!

The only thing I'm wondering is how reliable the AFK module is. On this [picture](https://pasteboard.co/JHVOoG6.png), you see that I locked my screen between 12:42 and 14:12 (indicated by LockApp.exe) and didn't touch the computer at this time. However, according to the AFK watcher I was on my computer until 13:34 and afterwards I was AFK with interruptions. Do you have any idea why these values are so far away from reality?

**Comment by johan-bjareholt** [comment, marked as answer]
Behavior like this has only been reported a couple of times the past few years and in both cases it was due to some USB device plugged in which reported small movements even when not touched (such as a gamepad joystick slightly drifting or a racing wheel being "too" accurate or simply just a cheap mouse).

Try plugging out all your USB devices and see if the AFK detection works correctly to know whether this is the issue or not. If it is, test plugging in/out different devices until you find the device which makes movements without you actually touching it.

**Reply by Culpeo** [reply]
Thanks for the answer, that sounds reasonable!
I have a ton of peripherals connected, so I'll do some tests and update this post with my findings.

**Reply by Culpeo** [reply]
It was the Xbox gamepad, which was connected via USB to the computer... I tested it several times now and the AFK module just shows this behaviour when the controller is plugged in.
Thank you very much for the quick and precise help!

**Comment by dogfuntom** [comment]
A wrote a workaround for those who are not ready to upgrade their mouses or controllers (or to unplug and replug them every 5 minutes): https://gist.github.com/dogfuntom/686e4ecf84f5b72eb4d700e98ce02227
